examples/service-clients/azure-client/todo-list/src/view.tsx

```diff
@@ -0,0 +1,205 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import {
+	CollaborativeInput,
+	CollaborativeTextArea,
+	SharedStringHelper,
+} from "@fluid-example/example-utils";
+import { useTree } from "@fluid-experimental/tree-react-api";
+import type { IFluidContainer } from "fluid-framework";
+import type { ISharedString, SharedString } from "fluid-framework/legacy";
+import React, { useEffect, useRef, useState } from "react";
+
+import { createTodoItem, type TodoListContainerSchema } from "./fluid.js";
+import type { TodoList, TodoItem } from "./schema.js";
+
+// eslint-disable-next-line import/no-unassigned-import
+import "./style.css";
+
+/**
+ * {@link TodoListAppView} input props.
+ */
+export interface TodoListAppViewProps {
+	readonly todoList: TodoList;
+	readonly container: IFluidContainer<TodoListContainerSchema>;
+}
+
+/**
+ * To-do list application view component.
+ */
+export const TodoListAppView: React.FC<TodoListAppViewProps> = (
+	props: TodoListAppViewProps,
+) => {
+	const { todoList, container } = props;
+
+	const [titleString, setTitleString] = useState<SharedString | undefined>();
+
+	const newItemTextInputRef = useRef<HTMLInputElement>(null);
+	useTree(todoList);
+
+	const todoListTitleHandle = todoList.title;
+
+	useEffect(() => {
+		todoListTitleHandle
+			.get()
+			.then((title) => {
+				setTitleString(title as ISharedString);
+			})
+			.catch((error) => {
+				console.error("Failed to get to-do list title:", error);
+				throw error;
+			});
+		return () => {};
+	}, [todoListTitleHandle]);
+
+	if (titleString === undefined) {
+		return <div>Loading...</div>;
+	}
+	const handleCreateClick = (ev: React.FormEvent<HTMLFormElement>): void => {
+		ev.preventDefault();
+
+		const input = newItemTextInputRef.current;
+		if (input === null) {
+			throw new Error("New item text field missing");
+		}
+
+		createTodoItem({
+			container,
+			initialTitleText: input.value,
+			completed: false,
+		})
+			.then((todoItem) => {
+				todoList.items.insertAtEnd(todoItem);
+				if (input !== null) {
```

Comment:
Consider grabbing the input.value and clearing the input above, prior to calling createTodoItem to avoid having to double-check the ref here.